### ISSUE TITLE
Miner.CompactPartitions doesn't reschedule replaced CC sectors correctly

### ISSUE THREAD

**Issue by wadeAlexC** [filecoin-project/specs-actors#1332]
When a CC sector is scheduled to be replaced, `ConfirmSectorProofsValid` reschedules its expiration to the end of its next deadline window:

https://github.com/filecoin-project/specs-actors/blob/638376f52995ac889abdc71c73082496c47ef25d/actors/builtin/miner/miner_actor.go#L818-L820

When these sectors are moved in `CompactPartitions`, they are placed back into an expiration queue at their declared expiration epoch.

**Comment by Stebalien** [comment]
Impact: a miner can end up causing a sector to live past when it should. Effectively "undoing" the upgrade. I believe the miner will even be able to retry to replace the sector, so this isn't too terrible (just wrong and annoying).

Possible solutions:

1. I'd kind of prefer to just update the sector info. It'll cost us something, but this issue has come up before.
2. An easier solution is to add some kind of flag. However, this is kind of hacky.
3. Check the first element in the expiration queue for an early expiration. This should be the only place where this can happen.
4. Pull expiration information from the previous partition's expiration queue instead of the sector info. This would be a massive refactor and I'd really hate to do this.

Thoughts on these proposals @anorth?

**Comment by anorth** [comment]
I'm aligned with updating the sector info. Let's try that first. It'll warrant a little thought to make sure that nothing is relying on the original value, but it's hard to imagine anything needing the "wrong" value.

**Comment by Stebalien** [comment]
Well, technically the sector termination fee, I think. But I think that's more of a bug at the moment anyways. That is, a miner could:

1. Replace.
2. Terminate.

And pay a fee assuming that the sector was only valid for 1 day. However, I'm pretty sure that's significantly  _more_ expensive than just:

1. Replace.
2. Fail to prove (let it go faulty). The expiration won't be moved.
3. Let it expire as faulty.

**Comment by anorth** [comment]
Right. After a successful replacement, I think a miner should only be on the hook for termination fees proportional to the reduced lifetime anyway (and letting it expire should still be cheaper).

**Comment by ZenGround0** [comment]
Since [FIP 0017](https://github.com/filecoin-project/FIPs/issues/131) is likely to land in v6 the current plan is to remove the old cc sector flow entirely as part of adding the new lightweight cc sectors.  This will fix this bug.
